### PR TITLE
Wallet API Testing

### PR DESCRIPTION
**Pull request by DavidVorick**
I cleaned up the documentation and server tester.

I switched from verifying batches of 50 blocks to verifying batches of 10 blocks because the database is slow and it causes problems to do it in batches of 50.

/wallet/seed PUT was changed to /wallet/address POST

Some of the api documentation was cleaned up.

The server tester object got functions for doing POST requests.

### CODE REVIEW

File: api/wallet_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+// wallet has never been encrypted.
+func TestIntegrationWalletGETEncrypted(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	// Check a wallet that has never been encrypted.
+	testdir := build.TempDir("api", "TestIntegrationWalletGETEncrypted")
+	APIAddr := ":" + strconv.Itoa(APIPort)
+	APIPort++
```
Comment (lukechampine) [inline api/wallet_test.go]: we're still using a global API port variable? I thought I got rid of that.

File: api/wallet_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		t.Error("there should not be unconfirmed incoming siacoins")
+	}
+
+	// Send coins to a wallet address through the api.
+	var wap WalletAddressPOST
+	err = st.postAPI("/wallet/address", url.Values{}, &wap)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sendSiacoinsValues := url.Values{}
```
Comment (lukechampine) [inline api/wallet_test.go]: generally prefer `var sendSiacoinsValues url.Values`

File: modules/consensus/synchronize_test.go [hunk 3]
```diff
@@ -171,7 +171,7 @@ func TestBlockHistory(t *testing.T) {
 	defer cst.closeCst()
 
 	// mine until we have enough blocks to test blockHistory
-	for cst.cs.Height() < MaxCatchUpBlocks {
+	for cst.cs.Height() < 50 {
```
Comment (lukechampine) [inline modules/consensus/synchronize_test.go]: why did changing `MaxCatchUpBlocks` break this?

File: modules/consensus/synchronize.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 const (
-	MaxCatchUpBlocks          = 50
+	MaxCatchUpBlocks          = 10
 	MaxSynchronizeAttempts    = 8
 	ResynchronizePeerTimeout  = time.Second * 30
 	ResynchronizeBatchTimeout = time.Minute * 3
```
Comment (lukechampine) [inline modules/consensus/synchronize.go]: these vars aren't used anymore

File: modules/consensus/synchronize.go [hunk 5]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 const (
-	MaxCatchUpBlocks          = 50
+	MaxCatchUpBlocks          = 10
```
Comment (lukechampine) [inline modules/consensus/synchronize.go]: what's the rationale here?

File: api/wallet_test.go (hunk 1 above)
Comment (DavidVorick) [inline api/wallet_test.go]: a ghost?

I just copy-pasted whatever had existed in the gateway test I think

File: modules/consensus/synchronize_test.go (hunk 3 above)
Comment (DavidVorick) [inline modules/consensus/synchronize_test.go]: If you read the rest of the test, the data is input manually to represent what would happen in a request of 50.

And it has nothing to do with MaxCatchUpBlocks lol, I'm not sure why the constant was used in the first place.

File: api/wallet_test.go (hunk 2 above)
Comment (DavidVorick) [inline api/wallet_test.go]: it's a map, doing `var ssv url.Values` causes nil map errors when you try to use it.

Comment (lukechampine) [inline api/wallet_test.go]: oh yeah. Confusing. I've never liked `url.Values`...usually it's easier to just write the encoded url by hand.

**Pull request merged**